src/libcore/ptr/non_null.rs

```diff
@@ -69,6 +67,8 @@ impl<T: Sized> NonNull<T> {
     #[rustc_const_stable(feature = "const_nonnull_dangling", since = "1.32.0")]
     #[inline]
     pub const fn dangling() -> Self {
+        // SAFETY: mem::align_of() returns a valid pointer. The caller has to
+        // handle the initialization of the pointer, and keep track of it.
```

Comment:
I think we only want to have as few information as possible in those comments. The only unsafe operation here is to create a `NonNull` value without checking for a null value, which is safe because `mem::align_of` returns a non-null pointer

Reply:
Fixed in aae8d54